src/lib/canvas/dropzone.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import styled from 'styled-components';
+import { map, filter, throttleTime, distinctUntilChanged } from 'rxjs/operators';
+import { fromEvent } from 'rxjs';
+import { SECTORS, SECTORS_HIGHLIGHT_COLOR, SECTORS_CENTER_RATIO } from './constants';
+import { getSectorCouple, isSmallSize } from './utils';
+
+const Container = styled.div`
+  height: 100%;
+  padding: 0;
+  margin: 0;
+  position:relative;
+`;
+
+const Sector = styled.div`
+  width: 50%;
+  height: 50%;
+  float: left;
+`;
+
+const Centre = styled.div`
+  position: absolute;
+  height: ${props => props.ratio * 100}%;
```

Comment:
props is not defined here. is it a feature of **styled** that allow you to use it here anyway?

Reply:
it's a feature of `styled-components`